core/common/src/ConcurrentQuery.ts

```diff
@@ -0,0 +1,410 @@
+  public setSuppressLogErrors(val: boolean) { this._config.suppressLogErrors = val; return this; }
+  public setLimit(val: QueryLimit) { this._config.limit = val; return this; }
+}
+/** @beta */
+export class BlobOptionsBuilder {
+  private _config: BlobOptions = {};
+  public get config(): BlobOptions { return this._config; }
+  public setPriority(val: number) { this._config.priority = val; return this; }
+  public setRestartToken(val: string) { this._config.restartToken = val; return this; }
+  public setQuota(val: QueryQuota) { this._config.quota = val; return this; }
+  public setUsePrimaryConnection(val: boolean) { this._config.usePrimaryConn = val; return this; }
+  public setRange(val: BlobRange) { this._config.range = val; return this; }
+}
+/** @internal */
+enum QueryParamType {
+  Boolean = 0,
+  Double = 1,
+  Id = 2,
+  IdSet = 3,
+  Integer = 4,
+  Long = 5,
+  Null = 6,
+  // eslint-disable-next-line @typescript-eslint/no-shadow
+  Point2d = 7,
+  // eslint-disable-next-line @typescript-eslint/no-shadow
+  Point3d = 8,
+  String = 9,
+  Blob = 10,
+  Struct = 11,
+}
+/** @beta */
+export class QueryBinder {
+  private _args = {};
+  private verify(indexOrName: string | number) {
+    if (typeof indexOrName === "number") {
+      if (indexOrName < 1)
+        throw new Error("expect index to be >= 1");
+    }
+    if (typeof indexOrName === "string") {
+      if (!/^[a-zA-Z_]+\w*$/i.test(indexOrName)) {
+        throw new Error("expect named parameter to meet identifier specification");
+      }
+    }
+  }
+  public bindBoolean(indexOrName: string | number, val: boolean) {
+    this.verify(indexOrName);
+    const name = String(indexOrName);
+    Object.defineProperty(this._args, name, {
+      enumerable: true,
+      value: {
+        type: QueryParamType.Boolean,
+        value: val,
+      },
+    });
+    return this;
+  }
+  public bindBlob(indexOrName: string | number, val: Uint8Array) {
+    this.verify(indexOrName);
+    const name = String(indexOrName);
+    const base64 = Buffer.from(val).toString("base64");
+    Object.defineProperty(this._args, name, {
+      enumerable: true, value: {
+        type: QueryParamType.Blob,
+        value: base64,
+      },
+    });
+    return this;
+  }
+  public bindDouble(indexOrName: string | number, val: number) {
+    this.verify(indexOrName);
+    const name = String(indexOrName);
+    Object.defineProperty(this._args, name, {
+      enumerable: true, value: {
+        type: QueryParamType.Double,
+        value: val,
+      },
+    });
+    return this;
+  }
+  public bindId(indexOrName: string | number, val: Id64String) {
+    this.verify(indexOrName);
+    const name = String(indexOrName);
+    Object.defineProperty(this._args, name, {
+      enumerable: true, value: {
+        type: QueryParamType.Id,
+        value: val,
+      },
+    });
+    return this;
+  }
+  public bindIdSet(indexOrName: string | number, val: Id64Set) {
```

Comment:
This should take an `OrderedId64Iterable` and call `CompressedId64Set.compressIds`.